src/control/server/storage/config.go

```diff
@@ -461,10 +461,11 @@ func (tcs TierConfigs) validateBdevRoles() error {
 // Role assignments will be decided based on the following rule set:
 //   - For 1 bdev tier, all roles will be assigned to that tier.
 //   - For 2 bdev tiers, WAL role will be assigned to the first bdev tier and Meta and Data to
```

Comment:
This is inaccurate. For 2 tiers, the combination could be "wal + meta" & "data" or "wal" & "meta + data", we just don't allow "wal + data" & "meta".

Reply:
Right, this comment is for `AssignBdevTierRoles()` function which is used to automatically assign roles during `dmg|daos_server config generate`. The comment "For 2 bdev tiers, WAL role will be assigned to the first bdev tier and Meta and Data to the second bdev tier." is describing the behaviour of the auto config population where when certain number of SSDs  have been detected for a given engine and the algorithm is trying to create a MD-on-SSD config then the distribution of roles across two tiers will be "wal" & "meta+data" and as you have described this is a valid combination. This isn't a validation function. Does that makes sense?

Reply:
For my understanding, to not have more than 3 bdev tiers with md-on-ssd is a limitation of the engine ?

Reply:
Yes. I see.

Reply:
@knard-intel This I have discussed with @NiuYawei and it seems unnecessary to be able to support that, it will be simpler for an administrator to understand and there doesn't seem to be any compelling use cases to support more than one data tier Rather than just adding any SSDs to a single data tier. From the perspective of the engine currently, each SSD is simply presented with a role that is assigned to the tier it came from, the tier ID is currently not used (although the information is present in the nvme.conf). If/when we move to differentiating between different SSD types TLC vs QLC and 3D NAND vs NAND then a use case for this may emerge. Also the situation where we have multiple data tiers in the form of e.g. "meta+data" and "data" it's not currently possible to distribute the roles engine side without changing the implementation. HTH

Reply:
Make sense, thanks for the explanation :)